opts.go

```diff
@@ -130,6 +135,13 @@ func (cache valueCache) cachedValue(
 	}
 
 	v, err := f()
-	cache[string(id)] = spliceValue{err, v}
+
+	// Only primitives can be cached, allowing us to get out of infinite loop
+	switch v.(type) {
+	case *cfgDynamic:
+	case cfgSub:
+	default:
+		cache[string(id)] = spliceValue{err, v}
+	}
```

Comment:
Don't review this part, I have to check on my side, if theses types are correct and would prefer white listing instead of black listing.

Reply:
whitelisting, like adding a `(value).canCache() bool` ?